test/extended/networking/isolation.go

```diff
@@ -1,35 +1,74 @@
 package networking
 
 import (
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/test/e2e"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
-// This test requires a network plugin that supports namespace isolation.
-// NOTE: if you change the test description, update networking.sh too!
 var _ = Describe("[networking] network isolation plugin", func() {
 	f1 := e2e.NewFramework("net-isolation1")
 	f2 := e2e.NewFramework("net-isolation2")
 
-	It("should prevent communication between pods in different namespaces", func() {
-		By("Picking multiple nodes")
-		nodes := getMultipleNodes(f1)
-		node1 := nodes.Items[0]
-		node2 := nodes.Items[1]
+	It("should prevent communication between pods in different namespaces on the same node", func() {
+		if pluginIsolatesNamespaces() {
+			checkPodIsolation(f1, f2, false)
+		} else {
+			e2e.Skipf("Network plugin does not implement isolation.")
```

Comment:
Consider defining constants (or a helper method) for implements/does not implement skip.